infrastructure/infrastructure.go

```diff
@@ -9,16 +12,24 @@ type InfraProvider interface {
 	GetInfraMachineTemplate(ctx context.Context, input GetInfraMachineTemplateInput) error
 
 	CreateInfraCluster(ctx context.Context, input CreateInfraClusterInput) error
-	CreateInfraStaticIdentity(ctx context.Context, input CreateInfraClusterStaticIdentityInput) error
+	CreateInfraStaticIdentity(ctx context.Context, input CreateInfraClusterIdentityInput) error
 	CreateInfraMachine(ctx context.Context, input CreateInfraMachineInput) error
 	CreateInfraMachineTemplate(ctx context.Context, input CreateInfraMachineTemplateInput) error
 
 	DeleteInfraCluster(ctx context.Context, input DeleteInfraClusterInput) error
-	DeleteInfraStaticIdentity(ctx context.Context, input DeleteInfraClusterStaticIdentityInput) error
+	DeleteInfraStaticIdentity(ctx context.Context, input DeleteInfraClusterIdentityInput) error
 	DeleteInfraMachine(ctx context.Context, input DeleteInfraMachineInput) error
 	DeleteInfraMachineTemplate(ctx context.Context, input DeleteInfraMachineTemplateInput) error
+
+	CreateClusterRoleIdentity(ctx context.Context, input CreateInfraClusterIdentityInput) error
+	GetClusterRoleIdentity(ctx context.Context, input GetClusterRoleIdentityInput) error
```

Comment:
```suggestion
	GetClusterRoleIdentity(ctx context.Context, input GetClusterIdentityInput) error
```
make this also generic

Reply:
done